src/analyzer/protocol/ldap/ldap_zeek.spicy

```diff
@@ -0,0 +1,13 @@
+module LDAP_Zeek;
+
+import LDAP;
+import zeek;
+
+on LDAP::ExtendedResponse::%done priority=-5 {
+  if ( ctx.messageMode == LDAP::MessageMode::TLS )
+    zeek::protocol_begin("SSL");
+}
+
+on LDAP::TlsForward::chunk {
+  zeek::protocol_data_in(zeek::is_orig(), self.chunk);
+}
```

Comment:
I find that having `ExtendedResponse` open the handle and `TlsForward` use it (via an implicit handle) to be confusing and potentially error prone. Is there are reason you couldn't open the handle on `TlsForward::%init`? If that is impossible I'd at least try to store the handle as a unit var and forward it to users like `TlsForward` as a unit parameter.

Reply:
I like the `%init` route, let me try.